### PR TITLE
refactor: remove nested walk function literal

### PR DESCRIPTION
**Pull request by jbedard**
This function literal confuses me everytime I see it and imo is nicer as a plain function.

**What type of PR is this?**
Other

**What package or component does this PR mostly affect?**
all

**What does this PR do? Why is it needed?**

Minor refactoring/cleanup.

### CODE REVIEW

**Comment by jbedard** [comment]
Make sure you look at the diff [ignoring whitespace](https://github.com/bazelbuild/bazel-gazelle/pull/1884/files?w=1)

File: walk/walk.go [hunk 1]
```diff
@@ -122,72 +122,72 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
...
-		if err != nil {
-			log.Print(err)
-			if c.Strict {
-				// TODO(https://github.com/bazelbuild/bazel-gazelle/issues/1029):
-				// Refactor to accumulate and propagate errors to main.
-				log.Fatal("Exit as strict mode is on")
-			}
-			haveError = true
-		}
+func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnoredFunc, knownDirectives map[string]bool, updateRels *UpdateFilter, wf WalkFunc, dir, rel string, updateParent bool) {
```
Comment (jbedard) [inline walk/walk.go:128]: The number of args here might be making it worth wrapping some of these fields in a struct?

Comment (fmeum) [reply walk/walk.go:128]: I actually prefer it this way, structs always make me suspicious of something else modifying it after the initial struct literal.

**Review by fmeum — APPROVED** [review]
Didn't know about `?w=1` on GitHub, this is super useful!

**Pull request merged**